src/libraries/System.IO/tests/StreamReader/StreamReaderTests.cs

```diff
@@ -110,6 +110,15 @@ public async Task ReadToEndAsync()
             Assert.Equal(5000, result.Length);
         }
 
+        [Fact]
+        public async Task ReadToEndAsync_WithCancellationToken()
```

Comment:
please add one more test that verifies that the operation is actually cancellable

you could do that by creating a large text file (100MB?), providing a cancellation token with short, but non-zero timeout like 10ms and verifying that an exception was thrown

Reply:
Assuming you mean a temp file here (not a `MemoryStream`); please let me know if that's not correct.